Assets/MultiBuild/Editor/SettingsWindow.cs

```diff
@@ -266,10 +283,10 @@ public class SettingsWindow : EditorWindow {
 
         void UpdateTargetsNotAdded() {
             _targetNamesNotAdded = new List<string>();
-            foreach (var target in Targets) {
-                if (!Settings.targets.Contains(target)) {
+            for (int i = 0, n = (int)Target.Max; i < n; ++i) {
+                Target target = Targets[i];
+                if (!Settings.targets.Contains(target))
```

Comment:
This should pickup every value in the `Targets` array. `Target.Max` is the count of enums inside `Target`. Here is a demonstration, all code is copied from the source of my commit: https://ideone.com/s2C15x
I weren't particularly trying to avoid the GC overhead of `foreach`, but rather switched because `foreach` wasn't properly iterating.